internal/pkg/cli/deploy/workload.go

```diff
@@ -370,24 +376,29 @@ func (d *workloadDeployer) uploadContainerImages(imgBuilderPusher imageBuilderPu
 
 func buildArgsPerContainer(name, workspacePath string, img ContainerImageIdentifier, unmarshaledManifest interface{}) (map[string]*dockerengine.BuildArguments, error) {
 	type dfArgs interface {
-		BuildArgs(rootDirectory string) map[string]*manifest.DockerBuildArgs
+		BuildArgs(rootDirectory string) (map[string]*manifest.DockerBuildArgs, error)
 		ContainerPlatform() string
 	}
 	mf, ok := unmarshaledManifest.(dfArgs)
 	if !ok {
 		return nil, fmt.Errorf("%s does not have required methods BuildArgs() and ContainerPlatform()", name)
 	}
-	var tags []string
-	tags = append(tags, imageTagLatest, img.ReferenceTag())
-	args := mf.BuildArgs(workspacePath)
-	dArgs := make(map[string]*dockerengine.BuildArguments, len(args))
-	for k, v := range args {
-		dArgs[k] = &dockerengine.BuildArguments{
-			Dockerfile: aws.StringValue(v.Dockerfile),
-			Context:    aws.StringValue(v.Context),
-			Args:       v.Args,
-			CacheFrom:  v.CacheFrom,
-			Target:     aws.StringValue(v.Target),
+	argsPerContainer, err := mf.BuildArgs(workspacePath)
+	if err != nil {
+		return nil, fmt.Errorf("check if manifest requires building from local Dockerfile: %w", err)
+	}
+	dArgs := make(map[string]*dockerengine.BuildArguments, len(argsPerContainer))
+	tags := []string{imageTagLatest, img.Tag()}
```

Comment:
nit: can we define `tags` inside of the for-loop to make it clear that each `tags` is at a container lvl